parser/src/dataset/mod.rs

```diff
@@ -501,12 +503,21 @@ where
                         match elem.into_value() {
                             Value::Primitive(_) | Value::PixelSequence { .. } => unreachable!(),
                             Value::Sequence(seq) => {
+                                let seq = if *invalidate_sq_len {
```

Comment:
@Enet4 I tried to make the actual change here. I _think_ that makes sense?

What I'm trying to do is conditionally make the `DataSetSequence` of undefined length, but I'm having a hard time. It seems not possible to change the `length`, and reconstructing seems to require some type arguments that I don't know how to provide :sweat_smile: 

```
error[E0283]: type annotations needed
   --> parser/src/dataset/mod.rs:507:70
    |
507 | ...                   DataSetSequence::new(seq.items().into(), Length::UNDEFINED)
    |                       --------------------             ^^^^
    |                       |
    |                       required by a bound introduced by this call
    |
    = note: cannot satisfy `_: Into<SmallVec<[I; 2]>>`
note: required by a bound in `dicom_core::value::DataSetSequence::<I>::new`
   --> /home/thomas/development/sandbox/dicom-rs/core/src/value/mod.rs:812:28
    |
812 |     pub fn new(items: impl Into<C<I>>, length: Length) -> Self {
    |                            ^^^^^^^^^^ required by this bound in `DataSetSequence::<I>::new`
help: try using a fully qualified path to specify the expected types
    |
507 |                                     DataSetSequence::new(<&[I] as Into<T>>::into(seq.items()), Length::UNDEFINED)
    |                                                          ++++++++++++++++++++++++           ~
    ```